news/8042.feature

```diff
@@ -0,0 +1,3 @@
+Add CLI option -H / --header for enhancing requests with HTTP headers.
+
+See https://github.com/pypa/pip/issues/8042 for discussion.
```

Comment:
The issue number will be attached automatically to the chngelog, so there’s no need to repeat the URL here. (hooray automation)